receiver/otlpreceiver/internal/trace/otlp.go

```diff
@@ -49,7 +51,30 @@ func (r *Receiver) Export(ctx context.Context, req ptraceotlp.ExportRequest) (pt
 	// NonPermanent errors will be converted to codes.Unavailable (equivalent to HTTP 503)
 	// Permanent errors will be converted to codes.InvalidArgument (equivalent to HTTP 400)
 	if err != nil {
-		return ptraceotlp.NewExportResponse(), errors.GetStatusFromError(err)
+		// See if an error is retryable.
+		cet := consumererror.Traces{}
+		if errors.As(err, &cet) {
+			retries := 3
+
+			// Retry three times; the real logic would be more complex akin to consumerretry.
+			for i := 0; i < retries; i++ {
+				// Re-use the data that was given to be retried.
+				td = cet.Data()
```

Comment:
We had discussed this before and determined that we want to take data from downstream since the data contained upstream is essentially just a pointer. In order for this to work possibly several components upstream with data that may have been mutated, we will need to find an unchanged copy for this to work.